Wikipedia/Code/WMFReadingThemesControlsViewController.swift

```diff
@@ -0,0 +1,171 @@
+    
+    var theme: Theme?
+    
+    @IBOutlet fileprivate var slider: SWStepSlider!
+    fileprivate var maximumValue: Int?
+    fileprivate var currentValue: Int?
+    
+    @IBOutlet weak var brightnessSlider: UISlider!
+    
+    @IBOutlet weak var standardThemeButton: UIButton!
+    @IBOutlet weak var lightThemeButton: UIButton!
+    @IBOutlet weak var darkThemeButton: UIButton!
+    
+    @IBOutlet weak var autoNightModeSwitch: UISwitch!
+    @IBOutlet weak var imageDimmingSwitch: UISwitch!
+    
+    @IBOutlet var separatorViews: [UIView]!
+    
+    @IBOutlet weak var minBrightnessImageView: UIImageView!
+    @IBOutlet weak var maxBrightnessImageView: UIImageView!
+    
+    @IBOutlet weak var tSmallImageView: UIImageView!
+    @IBOutlet weak var tLargeImageView: UIImageView!
+    
+    @IBOutlet var textLabels: [UILabel]!
+    
+    
+    var visible = false
+    
+    open weak var delegate: WMFReadingThemesControlsViewControllerDelegate?
+    
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+        if let max = self.maximumValue {
+            if let current = self.currentValue {
+                self.setValues(0, maximum: max, current: current)
+                self.maximumValue = nil
+                self.currentValue = nil
+            }
+        }
+        brightnessSlider.value = Float(UIScreen.main.brightness)
+        
+        // TODO: Enable when implemented
+        lightThemeButton.isEnabled = false
+        
+        autoNightModeSwitch.isEnabled = false
+        imageDimmingSwitch.isEnabled = false
+    }
+    
+    func applyBorder(to button: UIButton) {
+        button.borderColor = UIColor.wmf_blue
+        button.borderWidth = 2
+    }
+    
+    func removeBorderFrom(_ button: UIButton) {
+        button.borderWidth = 0
+    }
+    
+    open func setValuesWithSteps(_ steps: Int, current: Int) {
+        if self.isViewLoaded {
+            self.setValues(0, maximum: steps-1, current: current)
+        }else{
+            maximumValue = steps-1
+            currentValue = current
+        }
+    }
+    
+    func setValues(_ minimum: Int, maximum: Int, current: Int){
+        self.slider.minimumValue = minimum
+        self.slider.maximumValue = maximum
+        self.slider.value = current
+    }
+    
+    override open func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        visible = true
+        
+        if let name = UserDefaults.wmf_userDefaults().string(forKey: "WMFAppThemeName") {
+            switch name {
+            case "standard":
+                applyBorder(to: standardThemeButton)
+            case "light":
+                applyBorder(to: lightThemeButton)
+            case "dark":
+                applyBorder(to: darkThemeButton)
+            default:
+                break
+            }
+        }
+    }
+    
+    @IBAction func brightnessSliderValueChanged(_ sender: UISlider) {
+        UIScreen.main.brightness = CGFloat(sender.value)
```

Comment:
May want to also listen for `UIScreenBrightnessDidChangeNotification` and update the slider if the user changes the brightness using the system slider while this UI is up.